### ISSUE TITLE
AWS Health checks assume basic credentials

### ISSUE THREAD

**Issue by jocco** [Xabaril/AspNetCore.Diagnostics.HealthChecks#482]
**What would you like to be added**:
I would like to add a AWSCredentials field to `S3BucketOptions` and `DynamoDBOptions` replacing the current AccessKey and SecretKey string fields (I suggest they become deprecated and still handle that these are specified). These passed AWSCredentials would be used instead of the manual created BasicCredentials to call AWS.

This change could fallback to the default credentials if nothing is specified (thus supporting #119)

**Why is this needed**:

This is useful for cases were basic credentials are not available, for the full list of use cases see the AWS documentation - https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-config-creds.html.

Specifically for us we use AssumeRole credentials as well as AssumeRoleViaWebIdentity credentials to be able to assume IAM roles from our kubernetes service accounts. See https://docs.aws.amazon.com/eks/latest/userguide/specify-service-account-role.html

I am happy to provide a PR that resolves this.

**Comment by jocco** [comment]
If it was unclear what the intended changes would be here is a commit to my fork that would be roughly along the lines of what I was thinking (the deprecation of the old fields could be done in many different ways depending on what is prefered).

https://github.com/jocco/AspNetCore.Diagnostics.HealthChecks/commit/a417f1001e75e01fa45c993cf961cbf47924ecb2

**Comment by unaizorrilla** [comment]
Hi @jocco 

A PR for this is welcome, can you contribute this?

**Comment by gitfool** [comment]
None of our code provides any aws credentials at all since the AWS SDK for .NET [searches for credentials](https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-config-creds.html#creds-assign) anyway. In prod it just reads the EC2 instance metadata and when testing (inside containers) we inject the well-known environment variables.

TL;DR please allow for not specifying any credentials, thus leveraging the default credentials resolution.

**Comment by jocco** [comment]
Hey

Sorry for not providing the PR earlier, never got notified on your response.
Adding the PR now.

@gitfool not specifying credentials at all also seems like a valid option in terms of setup, it wouldn't work in my case though as we relying on the assume role work that AWS SDK does so simply removing the option to specify credentials wouldn't be a good solution for us.
